### PR TITLE
Vmimage cmd

### PR DESCRIPTION
**Pull request by richtja**
vmimage plugin for management vm images. Has two functions one for downloading new vmimages to the avocado cache and one for listing all existing vmimages inside avocado cache.

### CODE REVIEW

File: avocado/plugins/vmimage.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    :type version: str
+    :param arch: (optional) Architecture of the system image.
+    :type arch: str
+    """
+    cache_dirs = data_dir.get_cache_dirs()
+    image = vmimage.get(name=provider, version=version,
+                        arch=arch, cache_dir=cache_dirs)
+
+    LOG_UI.info("The image was downloaded:")
+    display_images_list([image])
```
Comment (apahim) [inline avocado/plugins/vmimage.py]: Line 58 below `says display_images_list()` is expecting a list of dicts. Is this line 50 even working?

File: avocado/plugins/vmimage.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+    def configure(self, parser):
+        parser = super(VMimage, self).configure(parser)
+        subparsers = parser.add_subparsers()
+        parser.add_argument('--list-downloaded',
+                            help='List of all downloaded images',
+                            action='store_true')
+        download_subcommand_parser = subparsers.add_parser(
+            'download', help='Downloads chosen VMimage')
+        download_subcommand_parser.add_argument('--provider',
```
Comment (apahim) [inline avocado/plugins/vmimage.py]: You meant distro, right? User should not care about provider...

File: selftests/unit/test_plugin_vmimage.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                        image['file'] = os.path.join(cache_dir, image['file'].
+                                                     format(version=provider.version,
+                                                            build=image['build'],
+                                                            arch=provider.arch).replace('$', ''))
+                        open(image["file"], "a").close()
+            expected_images = sorted(expected_images, key=lambda i: i['name'])
+            images = sorted(vmimage_plugin.list_downloaded_images(), key=lambda i: i['name'])
+            for index, image in enumerate(images):
+                for key in image:
+                    self.assertEqual(expected_images[index][key], image[key])
```
Comment (apahim) [inline selftests/unit/test_plugin_vmimage.py]: I'm missing functional tests for the CLI...

File: avocado/plugins/vmimage.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    Implements the avocado 'vmimage' subcommand
+    """
+
+    name = 'vmimage'
+    description = 'Provides VM images acquired from official repositories'
+
+    def configure(self, parser):
+        parser = super(VMimage, self).configure(parser)
+        subparsers = parser.add_subparsers()
+        parser.add_argument('--list-downloaded',
```
Comment (apahim) [inline avocado/plugins/vmimage.py]: Wondering what is the best interface here. Maybe:
````
$ avocado vmimage list
```
and 
```
$ avocado vmimage get
```
With options:
```
$ avocado vmimage get --arch ppc64 ...
```

File: avocado/plugins/vmimage.py (hunk 1 above)
Comment (richtja) [reply avocado/plugins/vmimage.py]: You are right, this is not working, I will write tests for catching those things. Thanks a lot.

File: avocado/plugins/vmimage.py (hunk 4 above)
Comment (richtja) [reply avocado/plugins/vmimage.py]: I like your idea I will change the interface.

File: avocado/utils/vmimage.py [hunk 5]
```diff
@@ -340,11 +365,15 @@ def __repr__(self):
                                                     self.arch)
 
     def get(self):
+        if isinstance(self.cache_dir, list):
+            cache_dirs = self.cache_dir
```
[the remark this answers is not in the document]
Comment (richtja) [reply avocado/utils/vmimage.py]: Good point, thanks.

**Comment by richtja** [comment]
Hi @apahim, I created new version #3269 with your comments. If you have time please have a look.

**Pull request closed without merging**